libsolidity/interface/CompilerStack.h

```diff
@@ -197,6 +197,10 @@ class CompilerStack: public langutil::CharStreamProvider
 		m_requestedContractNames = _contractNames;
 	}
 
+	/// Sets callback for reading from files or interactinng with SMT solvers.
+	/// Must be set before parsing.
+	void setCallback(ReadCallback::Callback _callback);
```

Comment:
We could do this, but I think it's better to just make the `CompilerStack` creation configurable in tests.

I just pushed a PR that does that and is more in line with how I'd like to change `AnalysisFramework`: #14473.